src/operator/contrib/krprod.h

```diff
@@ -0,0 +1,138 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ *  \file krprod.h
+ *  \brief Core function for Khatri-Rao product
+ *  \author Jencir Lee
+ */
+#ifndef MXNET_OPERATOR_CONTRIB_KRPROD_H_
+#define MXNET_OPERATOR_CONTRIB_KRPROD_H_
+#include <vector>
+#include "mshadow/tensor.h"
+
+namespace mxnet {
+namespace op {
+
+using namespace mshadow;
+using namespace mshadow::expr;
+
+/*!
+ * \brief Computes Khatri-Rao product
+ *
+ * Given the transpose of input matrices, this function computes the
+ * transpose of the Khatri-Rao product of the input matrices.
+ *
+ * Effectively, this implementation computes the Khatri-Rao product
+ * row by row, rather than column by column. E.g. if the input matrices
+ * are of shape (3, 2), (3, 4), (3, 5), the result matrix will be of
+ * shape (3, 2 * 4 * 5), which is (3, 40).
+ *
+ * \param out result matrix
+ * \param ts_arr vector of input matrices
+ */
+template <typename DType>
+inline void krprod
+  (Tensor<cpu, 2, DType> out,
+  const std::vector<Tensor<cpu, 2, DType> > &ts_arr) {
+  // If no input matrix, return all-one vector
```

Comment:
I see the point, we might not want to raise an error (a warning maybe?) but I still thing it should at least be null or empty. Otherwise it doesn't make as much sense and we makes potential bugs in the code hard to find.
What do you think?